### PR TITLE
Add infra for kjob repo.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
- Recent repo creation issue: https://github.com/kubernetes/org/issues/5279.
- Created repo https://github.com/kubernetes-sigs/kjob.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ presubmits:
     cluster: eks-prow-build-cluster
     branches:
     - ^main
-    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$|^(keps|cmd/experimental/kjobctl|site|charts)/"
```
Comment (mimowo) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:7]: Let's change Kueue in a follow up. Here we focus on kjob

Comment (mbobrovskyi) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml]: Done

**Comment by kannon92** [comment]
@mbobrovskyi considering your involvement in Kueue, I think you should consider becoming a member of the kubernetes-sig org.

File: config/jobs/kubernetes-sigs/kjob/OWNERS [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+- mimowo
+- mwielgus
+reviewers:
+- mimowo
+- mwielgus
+- mbobrovskyi
```
Comment (kannon92) [inline config/jobs/kubernetes-sigs/kjob/OWNERS:9]: To unblock this PR, you may need to @mbobrovskyi from reviewers. I still suggest you open up a request to join the kubernetes-sig org. Happy to sponsor you but we may want this in.

Comment (mbobrovskyi) [reply config/jobs/kubernetes-sigs/kjob/OWNERS]: Ah, I see. I've removed it for now.

**Comment by mimowo** [comment]
I don't think I can approve creation of the folder.
/assign @BenTheElder @dims

**Pull request merged**